tools/topology/platform/intel/intel-generic-dmic.m4

```diff
@@ -22,6 +22,10 @@ ifdef(`DMIC_DAI_LINK_48k_NAME',`',define(DMIC_DAI_LINK_48k_NAME, `dmic01'))
 # define(DMIC_DAI_LINK_16k_NAME, `dmic16k')
 ifdef(`DMIC_DAI_LINK_16k_NAME',`',define(DMIC_DAI_LINK_16k_NAME, `dmic16k'))
 
+# Need to define if not set
+ifdef(`DMIC_PCM_48k_ID',`', `define(DMIC_PCM_48k_ID, DMIC_DAI_LINK_48k_ID)')
+ifdef(`DMIC_PCM_16k_ID',`', `define(DMIC_PCM_16k_ID, DMIC_DAI_LINK_16k_ID)')
```

Comment:
this is done differently in intel-generic-dmic-kwd.m4

````
ifdef(`DMIC_PCM_48k_ID',`',`errprint(note: Need to define dmic48k pcm id for intel-generic-dmic-kwd
)')
````

I am not sure it makes sense to set a default here, in general device numbers and dailinks are really different items.

@ranj063 can you chime in?